artiq/test/coredevice/test_embedding.py

```diff
@@ -295,6 +303,7 @@ def test_args(self):
         self.assertEqual(exp.args1kwargs2(), 2)
         self.assertEqual(exp.numpy_things(),
                          (numpy.int32(10), numpy.int64(20), numpy.array([42,])))
+        self.assertEqual(type(exp.list_int64()[0]), type(numpy.int64(1)))
```

Comment:
Maybe also assert that the lists are equal (because why not), and add a comment, as this is slightly non-obvious? Something like
```
# Ensure lists of int64s don't decay to variable-length builtin integers.
list_int64 = exp.list_int64()
self.assertEqual(list_int64, self._list_int64)
self.assertTrue(isinstance(list_int64[0], numpy.int64))
```

Reply:
Could probably be a separate test case (`def test_int64_list(self): ...`) too, with the comment as a docstring.